src/core/Database.cpp

```diff
@@ -154,6 +154,20 @@ bool Database::open(const QString& filePath, QSharedPointer<const CompositeKey>
     setFilePath(filePath);
     dbFile.close();
 
+#ifdef Q_OS_WIN
+    // Get correct case for Windows filename (fixes #7139)
+    QFileInfo fileInfo(filePath);
+    QDir dir = fileInfo.dir();
+    QFileInfoList files = dir.entryInfoList();
```

Comment:
This could be simplified a bit
```
    QStringList files = dir.entryList(QStringList{"MAIN.cpp"}, QDir::Files);
    if (files.size() > 0) {
        setFilePath(files.at(0));
    }
```